### PR TITLE
Submissions now evaluate entity expressions and start one Cromwell workflow for each resulting entity

### PR DESCRIPTION
**Pull request by helgridly**

### CODE REVIEW

**Comment by helgridly** [comment]
Look at that. I even _increased_ coverage.

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphSubmissionDAO.scala [hunk 1]
```diff
@@ -69,11 +70,15 @@ class GraphSubmissionDAO extends SubmissionDAO with GraphDAO {
       val workspace = getWorkspaceVertex(db, workspaceNamespace, workspaceName).getOrElse(throw new IllegalArgumentException(s"workspace ${workspaceNamespace}/${workspaceName} does not exist"))
       val submissionVertex = setVertexProperties(submission, addVertex(db, null))
       workspace.addEdge(ExecutionEdgeTypes.submissionEdgeType, submissionVertex)
-      submission.workflow.foreach { workflow =>
+      submission.workflows.foreach { workflow =>
```
Comment (abaumann) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphSubmissionDAO.scala]: So this save creates new submission vertexes and edges each time right?  I guess the submission never changes, but we may want to throw an exception if you try to save more than once

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphSubmissionDAO.scala]: Submissions get a random uuid so this _shouldn't_ be a problem....

File: src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 2]
```diff
@@ -500,21 +501,56 @@ class WorkspaceService(dataSource: DataSource, workspaceDAO: WorkspaceDAO, entit
...
+                }
+
+                //Verify the type of all job entities matches the type of the method configuration.
+                if (!jobEntities.forall(_.entityType == methodConfig.rootEntityType)) {
+                  return RequestComplete(StatusCodes.BadRequest, "Entities " + jobEntities.filter(_.entityType != methodConfig.rootEntityType).map(e => e.entityType + ":" + e.name) +
+                    "are not the same type as the method configuration requires")
+                }
+
+                //Attempt to resolve method inputs and submit the workflows to Cromwell, and build the submission status accordingly.
+                val submittedWorkflows = jobEntities.map( e => submitWorkflow(workspaceName, methodConfig, e, wdl, authCookie, txn))
```
Comment (abaumann) [inline src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala]: what should happen with the submission when submitWorkflow fails?  Below we forall and the case is either Done or Submitted

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala]: I'm not sure I understand the question. The failures get collected into `notstarted` and the successes get collected into `workflows`; that information is returned with the submission.

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphSubmissionDAO.scala (hunk 1 above)
Comment (abaumann) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphSubmissionDAO.scala]: OK, we may just want to document that save saves a new copy each time, our other saves also update

File: src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala (hunk 2 above)
Comment (abaumann) [inline src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala]: this is resolved, we worked it out in person, just a misunderstanding on my part

**Pull request merged**